portal/patient_self_checkin/step5.php

```diff
@@ -0,0 +1,231 @@
+<?php
+$ignoreAuth = true;
+require_once "../../interface/globals.php";
+?>
+
+<!-- Latest compiled and minified CSS -->
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
+      integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u">
+
+<!-- Optional theme -->
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css"
+      integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp">
+
+<!-- Latest compiled and minified JavaScript -->
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"
+        integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa">
+
+</script>
+
+<!DOCTYPE html>
+<html>
+<head>
+    <title></title>
+</head>
+<body>
+
+<style>
+
+body {
+    font-family: sans-serif;
+    background-color: #638fd0;
+    background: -webkit-radial-gradient(circle, white, #638fd0);
+    background: -moz-radial-gradient(circle, white, #638fd0);
+  }
+
+  img {
+    text-align: center;
+}
+
+h1 {
+    text-align: center;
+}
+
+p {
+    text-align: center;
+}
+
+</style>
+
+<p>
+    <img style="text-align: center"
+         width="500px" src='<?php echo $GLOBALS['images_static_relative']; ?>/logo-full-con.png'/>
+</p>
+<!-- <img align="center" src="physician-icon-png-15318.png" width="100"><br> -->
+
+<?php
+
+$sex = htmlspecialchars($_GET['sex']);
+$month = htmlspecialchars($_GET['month']);
+$birthday = htmlspecialchars($_GET['date']);
+$surname = htmlspecialchars($_GET['surname']);
+
+// date_default_timezone_set();
+
+$timeRN = date("Y-m-d H:i:s");
+$date = date("y-m-d");
+echo "<h1> Time: " . $timeRN . "</h1><br>";
+
+require 'mySQL_connector.php';
+
+// Get the patient IDs of everyone with an appointment today. Enter them into an array which we can call upon later.
+
+$todaysPIDsQuery = "SELECT pc_pid from `openemr_postcalendar_events` WHERE `pc_eventDate` = '" . $date ."'";
+
+$todaysPIDs_result = mysqli_query($dbc, $todaysPIDsQuery); // This line executes the MySQL query that you typed above
+
+$PIDArray = array(); // make a new array to hold all your data
+
+$index = 0;
+while ($row = mysqli_fetch_array($todaysPIDs_result)) { // loop to store the data in an associative array.
+     $PIDArray[$index] = $row;
+     $index++;
+}
```

Comment:
The majority of the business logic here can be covered by several already existant functions. Will go through that tomorrow. But here's an example code block above that uses OpenEMR sql wrapper and binding for sql injection:
``````
$todaysPIDsQuery = "SELECT pc_pid from `openemr_postcalendar_events` WHERE `pc_eventDate` = ?";
$todaysPIDs_result = sqlStatement($todaysPIDsQuery, array($date));

$PIDArray = array(); // make a new array to hold all your data

$index = 0;
while ($row = sqlFetchArray($todaysPIDs_result)) { // loop to store the data in an associative array.
     $PIDArray[$index] = $row;
     $index++;
}
``````